src/190B20.c

```diff
@@ -1144,11 +1144,47 @@ INCLUDE_ASM(s32, "190B20", func_80263FE8);
 
 INCLUDE_ASM(s32, "190B20", func_80264084);
 
-INCLUDE_ASM(void, "190B20", add_xz_vec3f, Vec3f* vector, f32 speed, f32 angleDeg);
+void add_xz_vec3f(Vec3f* vector, f32 speed, f32 angleDeg) {
+    f32 newVectorZ;
+    f32 angleRad;
+    f32 sinAngleRad;
+
+    angleRad = angleDeg * 6.28318f;
```

Comment:
Here and below, the 6.28... can be `TAU`, and you can initialize angleRad on the line you declare it (along with putting it as the first variable declared):

`f32 angleRad = angleDeg * TAU / 360.0f;`

you can then declare and initialize sinAngleRad right after on the next line